### PR TITLE
feat: Add global auth env variables to `core.#Pull`

### LINKED ISSUE #2804: ✨ exceeded rate limiting for the docker hub
### What are you trying to do?

Users onboarding on Dagger sometimes need to login to their docker account in order to avoid triggering the rate limiting pull issues on the docker hub. However, most of our packages don't easily enable that.

Issue based on comments: https://github.com/dagger/dagger/issues/2238#issuecomment-1183530615

Solution: 
```
The way @helderco  suggested it was just to have two settings (e.g., DOCKERHUB_AUTH_USER, DOCKERHUB_AUTH_PASS) to apply in Go if they exist and the image to pull matches the index.docker.io registry, and has no auth.
```

Limitations: https://github.com/dagger/dagger/issues/2238#issuecomment-1183452408

Validated by @shykes: https://github.com/dagger/dagger/issues/2238#issuecomment-1183596368

### PR DESCRIPTION
**Pull request by grouville**
Add global `dockerhub_auth_user` and `dockerhub_auth_password` env variables, to enable login to docker hub (only) and avoid being limited by rate-limiting free accounts.

Fixes #2804 

Signed-off-by: guillaume

### CODE REVIEW

**Comment by grouville** [comment]
@calvinxu, this is your fix

**Comment by grouville** [comment]
Waiting for approval from @helderco and @shykes specifically (or one of the founders), on this one. Tested today with @calvinxu, it works fine

File: plan/task/pull.go
Comment (helderco) [inline plan/task/pull.go:62]: I'd use `os.LookupEnv` instead of looping the whole environ.

File: docs/guides/concepts/1244-docker.md
Comment (helderco) [inline docs/guides/concepts/1244-docker.md:71]: The registry address is `index.docker.io`.

File: tests/tasks.bats
Comment (helderco) [inline tests/tasks.bats:19]: You don't need to unset these vars. Bats does it for you (each test is isolated).

**Review by helderco — APPROVED** [review]
Left some minor comments.

Users may expect this to work also with `core.#Dockerfile` and `core.#Push` but we can cross that bridge when we get there.

File: plan/task/pull.go
Comment (grouville) [reply plan/task/pull.go:62]: I followed the current pattern in our codebase: https://github.com/dagger/dagger/blob/6a2368cf838d33cb1083ee01e29ac89b4846a0c3/plan/task/exec.go#L192-L218. If you prefer, I can change, I don't mind

Comment (grouville) [reply plan/task/pull.go:62]: Agreed

File: tests/tasks.bats
Comment (helderco) [reply tests/tasks.bats:19]: You marked as resolved but kept the line?

**Pull request merged**
